pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/FlinkPulsarSource.java

```diff
@@ -757,6 +764,21 @@ public void initializeState(FunctionInitializationContext context) throws Except
         }
     }
 
+    private MessageId checkAndLoadAvailableCursorState(String topic, MessageId messageId) {
```

Comment:
This logic is under discussion, we don't check the cursor status. Instead, we give user option to set their own cursor.